docs/headless-api.md

```diff
@@ -0,0 +1,120 @@
+
+**In order to keep our users safe from malicious plugins, RequestAccess must be the first function your plugin calls when using the Rojo API.** Attempting to use an API without requesting it beforehand will throw an error. The only exceptions to this are `Rojo.API.Version` and `Rojo.API.ProtocolVersion`, since those are useful in checking compatibility before anything else is done.
+
+Example Plugin:
+```Lua
+local granted = Rojo.API:RequestAccess(plugin, { "Connected", "ConnectAsync" })
+--[[
+granted = {
+	Connected = true, -- User granted access
+	ConnectAsync = false, -- User denied access
+}
+--]]
+```
+
+![image](https://user-images.githubusercontent.com/40185666/210909337-3caf5af4-0829-447c-9781-da3996c71284.png)
+
+
+
+```Lua
+Rojo.API:ConnectAsync(host: string?, port: string?): void
+```
+Attempts to connect Rojo to the given host and port. Same behavior as the user clicking the Connect button (sync locking, etc., all behave the same).
+
+```Lua
+Rojo.API:DisconnectAsync(): void
+```
+Attempts to disconnect any active sync session.
+
+```Lua
+Rojo.API:GetSetting(setting: string): any
+```
+Returns the value of the given setting.
+
+```Lua
+Rojo.API:SetSetting(setting: string, value: any): void
+```
+Sets the value of the given setting.
+
+```Lua
+function Rojo.API:Notify(msg: string, timeout: number?, actions: { [string]: {text: string, style: string, layoutOrder: number, onClick: (any) -> ()} }?): () -> ()
+```
+Sends a Rojo notification that indicates it comes from a third-party plugin. Returns a function that dismisses the notification.
+
+![image](https://github.com/boatbomber/rojo.space/assets/40185666/c3f31715-dcdd-4c9d-b947-8dceee606b61)
+
+
+```Lua
+Rojo.API:GetHostAndPort(): (string, string)
+```
+Returns the user's chosen host and port.
+
+```Lua
+Rojo.API:CreateApiContext(baseUrl: string): ApiContext
+```
+Returns a new ApiContext using the given baseUrl.
+
+### Properties *(All read-only)*
+
+```Lua
+Rojo.API.Version: {number}
+```
+The Rojo plugin version. (Example: `{7, 2, 1}`)
+
+```Lua
+Rojo.API.ProtocolVersion: number
+```
+The Rojo plugin's protocol version.
+
+```Lua
+Rojo.API.Connected: boolean
+```
+Whether Rojo is currently connected to a serve session.
+
+```Lua
+Rojo.API.Address: string?
+```
+When `Rojo.API.Connected` is `true`, this contains the address of the connected session.
+
+```Lua
+Rojo.API.ProjectName: string?
+```
+When `Rojo.API.Connected` is `true`, this contains the project name of the connected session.
+
+### Events
+
+```Lua
+Rojo.API.Changed: RbxScriptSignal (changedProperty: string, newValue: any?, oldValue: any?)
+```
+Fires when any property changes. Passes the name of the changed property and its new and old values.
```

Comment:
Can this be made clear that it's for properties of the Rojo API. I was confused when I first read it.